storage/innobase/log/log0log.cc

```diff
@@ -3375,6 +3381,11 @@ logs_empty_and_mark_files_at_shutdown(void)
 				(ulong) pending_io);
 			count = 0;
 		}
+		if (pending_io < pending_io_ref)
+		{
+			my_sd_extend_timeout(INNODB_EXTEND_TIMEOUT_INTERVAL);
+			pending_io_ref = pending_io;
+		}
```

Comment:
I think that this should be covered by instrumenting `buf_flush_remove()`. My understanding is that these are only referring to submitted but not yet completed asynchronous I/O requests. This should basically only kick in if the I/O subsystem is severely overloaded. And even then, tracking the real progress in `buf_flush_remove()` and possibly `log_write_up_to()` would be more helpful.